### PR TITLE
Add tool to extract glb from i3dm.

### PR DESCRIPTION
**Pull request by bagnell**

### CODE REVIEW

**Comment by pjcozzi** [comment]
Part of #34

**Comment by pjcozzi** [comment]
Looks good.  Can you add the instructions to [tools/README.md](https://github.com/AnalyticalGraphicsInc/3d-tiles-tools/blob/master/tools/README.md) and add tests?

File: tools/lib/i3dmToGlb.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+var getMagic = Cesium.getMagic;
+
+module.exports = i3dmToGlb;
+
+function i3dmToGlb(buffer) {
+    if (!defined(buffer)) {
+	throw new DeveloperError('buffer is not defined.');
```
Comment (lilleyse) [inline tools/lib/i3dmToGlb.js]: There are some mixed spaces vs. tabs in this file.

File: tools/lib/i3dmToGlb.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    var magicArray = new Uint8Array(4);
+    magicArray[0] = buffer.readUInt8(0);
+    magicArray[1] = buffer.readUInt8(1);
+    magicArray[2] = buffer.readUInt8(2);
+    magicArray[3] = buffer.readUInt8(3);
+
+    var magic = getMagic(magicArray, 0);
+    if (magic !== 'i3dm') {
+	throw new DeveloperError('i3dm is required.');
+    }
```
Comment (lilleyse) [inline tools/lib/i3dmToGlb.js]: I would just go with `var magic = buffer.toString('utf8', 0, 4);` instead.

File: tools/lib/i3dmToGlb.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    byteOffset += featureTableByteLength + batchTableByteLength;
+
+    var gltfByteLength = byteLength - byteOffset;
+    if (gltfByteLength === 0) {
+	throw new DeveloperError('glTF byte length is zero, i3dm must have a glTF to instance.');
+    }
+
+    var glbBuffer = new Buffer(gltfByteLength);
+    buffer.copy(glbBuffer, 0, byteOffset);
+    return glbBuffer;
```
Comment (lilleyse) [inline tools/lib/i3dmToGlb.js]: Slightly simpler with `Buffer.slice`.

File: tools/bin/3d-tiles-tools.js [hunk 4]
```diff
@@ -197,3 +205,20 @@ function readGlbWriteB3dm(inputPath, force, argv) {
                 });
         });
 }
+
+function readI3dmWriteGlb(inputPath, force, argv) {
+    var outputPath = defaultValue(
+        defaultValue(argv.o, argv.output),
+        defaultValue(argv._[2], inputPath.slice(0, inputPath.length - 3) + 'glb'));
```
Comment (lilleyse) [inline tools/bin/3d-tiles-tools.js]: Should be `- 4`

**Comment by pjcozzi** [comment]
Bump @bagnell.

**Comment by bagnell** [comment]
@pjcozzi @lilleyse This is ready for another look.

**Comment by pjcozzi** [comment]
Works great, thanks!

**Pull request merged**